bin/p2-rm/main.go

```diff
@@ -40,37 +67,123 @@ func main() {
 		*nodeName = hostname
 	}
 
-	// If the node is managed by a controller, let the user know. Changes to the
-	// controller might be needed.
-	labelID := path.Join(*nodeName, *podName)
-	podLabels, err := labeler.GetLabels(labels.POD, labelID)
+	rm := NewP2RM(kp.NewConsulClient(opts), *podName, *nodeName)
+
+	podIsManagedByRC, rcID, err := rm.checkForManagingReplicationController()
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "unable to check node for labels: %v\n", err)
-		os.Exit(1)
+		os.Exit(2)
 	}
-	if podLabels.Labels.Has(rc.RCIDLabel) {
+
+	if !podIsManagedByRC {
+		err = rm.deletePod()
+		if err != nil {
+			os.Exit(2)
+		}
+	}
+
+	if podIsManagedByRC && !*deallocation {
 		fmt.Fprintf(
 			os.Stderr,
-			"warning: %s is managed by replication controller: %v\n"+
-				"         To reduce the controller's capacity, use \"p2-rctl set-replicas\"\n"+
-				"continue [y|n]? ",
-			labelID,
-			podLabels.Labels.Get(rc.RCIDLabel),
-		)
-		input, err := bufio.NewReader(os.Stdin).ReadString('\n')
-		answer := (strings.ToLower(strings.TrimSpace(input)) + "n")[0]
-		if err != nil || answer != 'y' {
-			fmt.Fprintf(os.Stderr, "aborting\n")
+			"error: %s is managed by replication controller: %v\n"+
+				"It's possible you meant you deallocate this pod on this node. If so, please confirm your intention with --deallocate\n", rcID)
+		os.Exit(2)
```

Comment:
Good to know, but I'm not keen to add a dependency on the log package here. Right now we're doing human friendly printing to stderr